src/analyses/interval_domain.h

```diff
@@ -56,26 +56,40 @@ class interval_domaint:public ai_domain_baset
   }
 
   // no states
-  void make_bottom() final
+  void make_bottom() final override
   {
     int_map.clear();
     float_map.clear();
     bottom=true;
   }
 
   // all states
-  void make_top() final
+  void make_top() final override
   {
     int_map.clear();
     float_map.clear();
     bottom=false;
   }
 
-  void make_entry() final
+  void make_entry() final override
   {
     make_top();
   }
 
+  bool is_bottom() const override final
+  {
+    // This invariant should hold but is not correctly enforced at the moment.
```

Comment:
could this be marked as TODO to be easily found? or is there already an issue?

Reply:
There isn't directly an issue because there are many things wrong with the interval domain.  For example, the majority of the transformers are not implemented, it will produce unsound results in the presence of writes to pointers, etc.  The goal is to implement an interval abstraction within the variable sensitivity domain which doesn't have these design-level problems and then deprecate this.  There are various PRs for that and I am reluctant to spend too much more time on this interval domain as it cannot be developed to the level of correctness that the other one can.